source/particle/property/elastic_tensor_decomposition.cc

```diff
@@ -0,0 +1,479 @@
+            dealii::Tensor<1,21> mono_and_higher_vector;
+            mono_and_higher_vector[0] = full_elastic_vector_rotated[0];
+            mono_and_higher_vector[1] = full_elastic_vector_rotated[1];
+            mono_and_higher_vector[2] = full_elastic_vector_rotated[2];
+            mono_and_higher_vector[3] = full_elastic_vector_rotated[3];
+            mono_and_higher_vector[4] = full_elastic_vector_rotated[4];
+            mono_and_higher_vector[5] = full_elastic_vector_rotated[5];
+            mono_and_higher_vector[6] = full_elastic_vector_rotated[6];
+            mono_and_higher_vector[7] = full_elastic_vector_rotated[7];
+            mono_and_higher_vector[8] = full_elastic_vector_rotated[8];
+            mono_and_higher_vector[11] = full_elastic_vector_rotated[11];
+            mono_and_higher_vector[14] = full_elastic_vector_rotated[14];
+            mono_and_higher_vector[17] = full_elastic_vector_rotated[17];
+            mono_and_higher_vector[20] = full_elastic_vector_rotated[20];
+
+            auto tric_vector = full_elastic_vector_rotated-mono_and_higher_vector;
+            norms[0][permutation_i] = tric_vector.norm_square();
+
+            // The following line would do the same as the lines below, but it is slow. It has therefore been
+            // replaced by the lines below.
+            //auto ortho_and_higher_vector = projection_matrix_mono_to_ortho*mono_and_higher_vector;
+            dealii::Tensor<1,9>  mono_and_higher_vector_cropped;
+            mono_and_higher_vector_cropped[0] = mono_and_higher_vector[0];
+            mono_and_higher_vector_cropped[1] = mono_and_higher_vector[1];
+            mono_and_higher_vector_cropped[2] = mono_and_higher_vector[2];
+            mono_and_higher_vector_cropped[3] = mono_and_higher_vector[3];
+            mono_and_higher_vector_cropped[4] = mono_and_higher_vector[4];
+            mono_and_higher_vector_cropped[5] = mono_and_higher_vector[5];
+            mono_and_higher_vector_cropped[6] = mono_and_higher_vector[6];
+            mono_and_higher_vector_cropped[7] = mono_and_higher_vector[7];
+            mono_and_higher_vector_cropped[8] = mono_and_higher_vector[8];
+            dealii::Tensor<1,9> ortho_and_higher_vector;
+            ortho_and_higher_vector[0] = mono_and_higher_vector[0];
+            ortho_and_higher_vector[1] = mono_and_higher_vector[1];
+            ortho_and_higher_vector[2] = mono_and_higher_vector[2];
+            ortho_and_higher_vector[3] = mono_and_higher_vector[3];
+            ortho_and_higher_vector[4] = mono_and_higher_vector[4];
+            ortho_and_higher_vector[5] = mono_and_higher_vector[5];
+            ortho_and_higher_vector[6] = mono_and_higher_vector[6];
+            ortho_and_higher_vector[7] = mono_and_higher_vector[7];
+            ortho_and_higher_vector[8] = mono_and_higher_vector[8];
+            auto mono_vector = mono_and_higher_vector_cropped-ortho_and_higher_vector;
```

Comment:
can you change the variable names to always contain the full name of the crystal system? `mono` is a very ambiguous name, `monoclinal` is much more specific.